sklearn/metrics/_classification.py

```diff
@@ -141,6 +147,28 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _nan_average(scores: np.ndarray, weights: Optional[np.ndarray]):
```

Comment:
From the code it looks like `weights` can also be a list. I prefer not to have type hints in this file to be consistent.